gnomad/utils/constraint.py

```diff
@@ -360,3 +412,291 @@ def collapse_strand(
+    :param possible_variants_expr: Int64Expression of the possible variant counts.
+    :param pops_observed_variants_array_expr: Nested ArrayExpression with all observed
+        variant counts ArrayNumericExpressions for specified populations. e.g., `[[1,1,
+        1],[1,1,1]]`. Default is None.
+    :param weighted: Whether to generalize the model to weighted least squares using
+        'possible_variants'. Default is False.
+    :return: A dictionary of intercepts and slopes of plateau models for all sites and
+        populations (optional). The keys for this dictionary are 'total' and 'pop'
+        (optional). The values for 'total' is a dictionary (e.g., <DictExpression of
+        type dict<bool,array<float64>>>), and the value for 'pop' is a nested list of
+        dictionaries (e.g., <ArrayExpression of type array<array<dict<bool,
+        array<float64>>>>>). The key of the dictionary in the nested list is CpG status
+        (BooleanExpression), and the value is an ArrayExpression containing intercept
+        and slope values.
+    """
+    # Build plateau models for all sites
+    plateau_models_agg_expr = {
+        "total": hl.agg.group_by(
+            cpg_expr,
+            hl.agg.linreg(
+                observed_variants_expr / possible_variants_expr,
+                [1, mu_snp_expr],
+                weight=possible_variants_expr if weighted else None,
+            ).beta,
+        )
+    }
+    if pops_observed_variants_array_expr:
+        # Build plateau models using sites in population downsamplings if
+        # population is specified.
+        plateau_models_agg_expr["pop"] = hl.agg.array_agg(
+            lambda pop_obs_var_array_expr: hl.agg.array_agg(
+                lambda pop_observed_variants: hl.agg.group_by(
+                    cpg_expr,
+                    hl.agg.linreg(
+                        pop_observed_variants / possible_variants_expr,
+                        [1, mu_snp_expr],
+                        weight=possible_variants_expr,
+                    ).beta,
+                ),
+                pop_obs_var_array_expr,
+            ),
+            pops_observed_variants_array_expr,
+        )
+    return plateau_models_agg_expr
+
+
+def build_coverage_model(
+    low_coverage_oe_expr: hl.expr.Float64Expression,
+    log_coverage_expr: hl.expr.Float64Expression,
+) -> hl.expr.StructExpression:
+    """
+    Build coverage model.
+
+    This function uses linear regression to build a model of log10(coverage) to correct
+    proportion of expected variation at low coverage sites.
+
+    The x and y of the coverage model:
+    - x: log10('exome_coverage') at low coverage site
+    - y: sum('observed_variants')/ (`high_coverage_scale_factor` * sum('possible_variants' * 'mu_snp') at low coverage sites
+    where `high_coverage_scale_factor` = sum('observed_variants') / sum('possible_variants' * 'mu_snp') at high coverage sites
```

Comment:
```suggestion
    - x: `log_coverage_expr`
    - y: low_coverage_oe_expr`
```
(editing this formula for this function)